src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs

```diff
@@ -220,18 +222,11 @@ public async Task<IEnumerable<(string ItemId, string[] ItemMetadata)>> GetBuildI
                 throw new ArgumentNullException(nameof(itemName));
             }
 
-            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-            var itemStorage = VsHierarchy as IVsBuildItemStorage;
-            if (itemStorage != null)
```

Comment:
I'm in two minds about this change. Let's assume someone is building a new project system, they make NuGet think it supports NuGet and PackageReference, but isn't a CPS project so ends up here, but most importantly the project doesn't implement `IVsBuildItemStorage`.

Before this change, NuGet will silently ignore the failure to get items, always telling the rest of our code "no, this project doesn't define any items of that type".  If the project does in fact define items of that type, then this will be **EXTREMELY** difficult for us to diagnose.

With my change, the code will throw an exception telling us exactly what the problem is. However, it will throw even when the project doesn't define items with the specified name, and therefore this codepath isn't even needed. This will completely break NuGet with this project system, making it impossible to restore, even if the customer tries to avoid using CPM or NuGetAuditSuppress (the only two places this method is used currently).

What do others think?

Reply:
We don't really know about the project systems there and whether they implement IVsBuildItemStorage. 

Maybe I'm more paranoid, but I'm leaning towards having a check, because we're likely to only see these issues *after* we ship. 

You can maybe feature flag it or something, if you want to do an experiment.